router/batchrouter/batchrouter.go

```diff
@@ -2374,3 +2378,18 @@ func (brt *HandleT) updateRudderSourcesStats(ctx context.Context, tx jobsdb.Upda
 	}
 	return nil
 }
+
+func (brt *HandleT) updateProcessedEventsMetrics(statusList []*jobsdb.JobStatusT) {
+	countPerState := map[string]int{}
+	for i := range statusList {
+		state := statusList[i].JobState
+		countPerState[state]++
+	}
+	for state, count := range countPerState {
+		stats.NewTaggedStat(`pipeline_processed_events`, stats.CountType, stats.Tags{
+			"module":   "batch_router",
+			"destType": brt.destType,
+			"state":    state,
```

Comment:
[question] are we only capturing terminal states here?

Reply:
I understand we can capture non-terminal states as well. What if we introduce a boolean flag `terminal`.

This will give us the option to measure flow in/ out without the noise of retries. If we want to know that the system is working, then we can filter without that flag

Reply:
we capture any state, if we want to measure terminal states, it is possible, we can add a state filter in the query